File: C version/141.Linked-list-cycle.c
#include <stdio.h>
#include <stdlib.h>

#define TRUE 1
#define FALSE 0

struct ListNode{
    int val;
    struct ListNode *next;
};

int hasCycle(struct ListNode *head){
    struct ListNode *ptrFast;
    struct ListNode *ptrSlow;

    if(head == NULL || head->next == NULL){
        return FALSE;
    }

    ptrFast = head;
    ptrSlow = head;

    while(ptrFast != NULL && ptrFast->next != NULL){
        ptrFast = ptrFast->next->next;
        ptrSlow = ptrSlow->next;
        if(ptrFast == ptrSlow){
            return TRUE;
        }
    }
    return FALSE;
}

File: 141.linked-list-cycle.java

class ListNode {
    int val;           
    ListNode next;     
    
    ListNode() {}   
    
    ListNode(int val) { 
        this.val = val; 
    }
    
    ListNode(int val, ListNode next) {
        this.val = val; 
        this.next = next; 
    }
}


class Solution {
    public boolean hasCycle(ListNode head) {
        if(head == null || head.next == null){
            return false;
        }

        ListNode ptrFast = head;
        ListNode ptrSlow = head;

        while(ptrFast != null && ptrFast.next != null){
            ptrFast = ptrFast.next.next;
            ptrSlow = ptrSlow.next;
            if(ptrFast == ptrSlow){
                return true;
            }
        }
        return false;
    }
}